packages/hierarchies-react/src/test/UseSelectionHandler.test.tsx

```diff
@@ -0,0 +1,417 @@
+        await user.keyboard(`{/Shift}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1", "node-2", "node-3"], "replace");
+      });
+
+      it("skips info nodes when selecting range", async () => {
+        const nodes = [createHierarchyNode("node-1"), createInfoNode("node-2"), createHierarchyNode("node-3")];
+        const { user, getByText } = render(<TestComponent {...createProps(nodes, selectionMode, true)} />);
+
+        const node1 = getByText("node-1");
+        node1.focus();
+        await clickNode(user, node1);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1"], "replace");
+        selectNodesStub.reset();
+
+        const node3 = getByText("node-3");
+        node3.focus();
+
+        await user.keyboard(`{Shift>}`);
+        await clickNode(user, node3);
+        await user.keyboard(`{/Shift}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1", "node-3"], "replace");
+      });
+
+      it("selects visible children of different depth when selecting range", async () => {
+        const innerChild = createHierarchyNode("child-inner");
+        const outerChild = createHierarchyNode("child-outer", [innerChild]);
+        const nodes = [createHierarchyNode("node-1", [outerChild]), createHierarchyNode("node-2")];
+        const { user, getByText } = render(<TestComponent {...createProps(nodes, selectionMode, true)} />);
+
+        const node1 = getByText("node-1");
+        node1.focus();
+        await clickNode(user, node1);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1"], "replace");
+        selectNodesStub.reset();
+
+        const node2 = getByText("node-2");
+        node2.focus();
+
+        await user.keyboard(`{Shift>}`);
+        await clickNode(user, node2);
+        await user.keyboard(`{/Shift}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1", "child-outer", "child-inner", "node-2"], "replace");
+      });
+
+      it("skips non visible children when selecting range", async () => {
+        const innerChild = createHierarchyNode("child-inner");
+        const outerChild = createHierarchyNode("child-outer", [innerChild]);
+        const nodes = [createHierarchyNode("node-1", [outerChild], false), createHierarchyNode("node-2")];
+        const { user, getByText } = render(<TestComponent {...createProps(nodes, selectionMode, true)} />);
+
+        const node1 = getByText("node-1");
+        node1.focus();
+        await clickNode(user, node1);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1"], "replace");
+        selectNodesStub.reset();
+
+        const node2 = getByText("node-2");
+        node2.focus();
+
+        await user.keyboard(`{Shift>}`);
+        await clickNode(user, node2);
+        await user.keyboard(`{/Shift}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1", "node-2"], "replace");
+      });
+
+      it("does not update previous selection when using `shift`", async () => {
```

Comment:
Test does many selection changes and it is not clear from test name what it checks. Can you update test name and clarify what is the goal of it?

Reply:
Renamed the test.